test/validate_test.go

```diff
@@ -0,0 +1,277 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/terraform-validator/converters/google"
+	"github.com/GoogleCloudPlatform/terraform-validator/tfgcv"
+	"github.com/GoogleCloudPlatform/terraform-validator/tfplan"
+	"github.com/hashicorp/terraform/terraform"
+	"google.golang.org/api/cloudresourcemanager/v1"
+	"google.golang.org/api/option"
+)
+
+func TestValidate(t *testing.T) {
+	cfg := configure(t)
+	ancestry := "organization/example.com/folder/foobar"
+	resourceManager, err := cloudresourcemanager.NewService(context.Background(), option.WithoutAuthentication())
+	if err != nil {
+		t.Fatalf("constructing resource manager client: %v", err)
+	}
+	baseResourceInstanceDiff := &terraform.InstanceDiff{
+		Attributes: map[string]*terraform.ResourceAttrDiff{
+			"name": &terraform.ResourceAttrDiff{New: "new-foo"},
+		},
+	}
+
+	type testcase struct {
+		kind string
+		name string
+		diff *terraform.InstanceDiff
```

Comment:
Is this output or input for the testcase?  Can you comment on what is getting tested against what?